pos_ticket_static_qrcode/models/pos_config.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2022 Moka Tourisme (https://www.mokatourisme.fr).
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+from odoo import api, fields, models
+import qrcode
+from qrcode.image import pil
+from qrcode.constants import ERROR_CORRECT_M
+import io
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    pos_ticket_static_qrcode_value = fields.Char(
+        string="QRCode Value",
+        help=(
+            "QRCode value use to generate QRCOde, "
+            "the value return when user scan that QRCode."
+        )
+    )
+    pos_ticket_static_qrcode_message = fields.Char(
+        string="QRCode Message",
+        help=(
+            "Message to display under QRCode."
+        )
+    )
+    pos_ticket_static_qrcode = fields.Binary(
+        string="QRCode", compute="_compute_qr_code", readonly=True
+    )
+
+    @api.depends("pos_ticket_static_qrcode_value")
+    def _compute_qr_code(self):
+        """Mainly inspired from report_qr
+
+        I'm not using the report qrcode engine because that qrcode
+        is loaded at the begining of the POS session.
+        """
+        for config in self:
+            if config.pos_ticket_static_qrcode_value:
+                try:
+                    qr = qrcode.QRCode(
+                        box_size=3,
+                        border=5,
+                        image_factory=pil.PilImage,
+                        error_correction=ERROR_CORRECT_M,
+                    )
+                    qr.add_data(self.pos_ticket_static_qrcode_value)
+                    qr.make()
+                    img = qr.make_image(
+                        fill_color="white", back_color="black"
+                    )
+                    png = io.BytesIO()
+                    img.save(png)
+                    png.seek(0)
+                except Exception:
+                    raise ValueError("Cannot convert into barcode.")
```

Comment:
Into QR code, i guess ?

Reply:
Thanks for your review I've improve this message and the exception is now covered by unittest !